### ISSUE TITLE
Vault Constructor Optional Fields

### ISSUE THREAD

**Issue by johnnysinger** [kyndryl-open-source/hashi-vault-js#33]
**Describe the bug**
When constructing the Vault in typescript there are optional fields that are required.

**To Reproduce**
1. Install hashi-vault-js `yarn add hashi-vault-js`
2. Initiate object this way 

```
import Vault, { } from 'hashi-vault-js';

const vault = new Vault({
      https: true,
      baseUrl: `${process.env.VAULT_ADDR}/v1`,
      rootPath: 'secret',
      timeout: 2000,
      proxy: false,
      namespace: process.env.VAULT_NAMESPACE
    });
```

4. You will see the type error saying  cert,key,cacert are required

**Expected behavior**
we should be able to not include those fields if they are optional

**Environment:**
* Node.js version:  18.7.1
* hashi-vault-js version: 0.4.12

**Additional context**
The current workaround for this is empty strings 

```
import Vault, { } from 'hashi-vault-js';

const vault = new Vault({
      https: true,
      baseUrl: `${process.env.VAULT_ADDR}/v1`,
      rootPath: 'secret',
      cert: '',
      key: '',
      cacert: '',
      timeout: 2000,
      proxy: false,
      namespace: process.env.VAULT_NAMESPACE
    });
```

**Comment by rod-anami** [comment]
Hmm...this seems to be a known issue as we only updated the JavaScript constructor properly.

**Comment by phr3nzy** [comment]
https://github.com/rod4n4m1/hashi-vault-js/pull/35

**Comment by rod-anami** [comment]
This one has been addressed on #35. Tagged for `0.4.13`.
